src/sample/skinnedMesh/main.ts

```diff
@@ -0,0 +1,570 @@
+      modelMatrix.byteOffset,
+      modelMatrix.byteLength
+    );
+
+    // Write to skinned grid bone uniform buffer
+    for (let i = 0; i < gridBoneCollection.transforms.length; i++) {
+      device.queue.writeBuffer(
+        skinnedGridJointUniformBuffer,
+        i * 64,
+        gridBoneCollection.transforms[i] as Float32Array
+      );
+    }
+
+    // Difference between these two render passes is just the presence of depthTexture
+    gltfRenderPassDescriptor.colorAttachments[0].view = context
+      .getCurrentTexture()
+      .createView();
+
+    skinnedGridRenderPassDescriptor.colorAttachments[0].view = context
+      .getCurrentTexture()
+      .createView();
+
+    // Update node matrixes
+    for (const scene of whaleScene.scenes) {
+      scene.root.updateWorldMatrix(device);
+    }
+
+    // Updates skins (we index into skins in the renderer, which is not the best approach but hey)
+    animWhaleSkin(whaleScene.skins[0], Math.sin(t) * settings.angle);
+    // Node 6 should be the only node with a drawable mesh so hopefully this works fine
+    whaleScene.skins[0].update(device, 6, whaleScene.nodes);
+
+    const commandEncoder = device.createCommandEncoder();
+    if (settings.object === 'Whale') {
+      const passEncoder = commandEncoder.beginRenderPass(
+        gltfRenderPassDescriptor
+      );
+      for (const scene of whaleScene.scenes) {
+        scene.root.renderDrawables(passEncoder, [
+          cameraBGCluster.bindGroups[0],
+          generalUniformsBGCLuster.bindGroups[0],
+        ]);
+      }
+      passEncoder.end();
+    } else {
+      const passEncoder = commandEncoder.beginRenderPass(
+        skinnedGridRenderPassDescriptor
+      );
+      passEncoder.setPipeline(skinnedGridPipeline);
+      passEncoder.setBindGroup(0, cameraBGCluster.bindGroups[0]);
+      passEncoder.setBindGroup(1, generalUniformsBGCLuster.bindGroups[0]);
+      passEncoder.setBindGroup(2, skinnedGridBoneBGCluster.bindGroups[0]);
+      passEncoder.setVertexBuffer(0, skinnedGridVertexBuffers.positions);
+      passEncoder.setVertexBuffer(1, skinnedGridVertexBuffers.joints);
+      passEncoder.setVertexBuffer(2, skinnedGridVertexBuffers.weights);
+      passEncoder.setIndexBuffer(skinnedGridVertexBuffers.indices, 'uint16');
+      passEncoder.drawIndexed(gridIndices.length, 1);
+      passEncoder.end();
+    }
+
+    device.queue.submit([commandEncoder.finish()]);
+
+    requestAnimationFrame(frame);
+  }
+  requestAnimationFrame(frame);
+};
+
+const skinnedMesh: () => JSX.Element = () =>
+  makeSample({
+    name: 'Skinned Mesh',
+    description:
+      'A demonstration of basic gltf loading and mesh skinning, ported from https://webgl2fundamentals.org/webgl/lessons/webgl-skinning.html. Mesh data, per vertex attributes, and skin inverseBindMatrices are taken from the json parsed from the binary output of the .glb file, with animated joint matrices updated and passed to shaders per frame via uniform buffers.',
```

Comment:
To avoid having readers make the same mistake I did, maybe update this to say something like:

```suggestion
      'A demonstration of basic gltf loading and mesh skinning, ported from https://webgl2fundamentals.org/webgl/lessons/webgl-skinning.html. Mesh data, per vertex attributes, and skin inverseBindMatrices are taken from the json parsed from the binary output of the .glb file. Animations are generated programmatically with animated joint matrices updated and passed to shaders per frame via uniform buffers.',
```

Reply:
Changed